flink-core/src/main/java/org/apache/flink/configuration/DescribedEnum.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.configuration;
+
+import org.apache.flink.annotation.PublicEvolving;
+import org.apache.flink.configuration.description.Description;
+import org.apache.flink.configuration.description.InlineElement;
+import org.apache.flink.configuration.description.TextElement;
+
+import java.util.List;
+
+/**
+ * Describe enum constants used in {@link ConfigOption}s.
+ *
+ * <p>For enums used as config options, this interface can be implemented to provide a {@link
+ * Description} for each enum constant. This will be used when generating documentation for config
+ * options to include a list of available values alongside their respective descriptions.
+ *
+ * <p>More precisely, only a list of {@link InlineElement}s can be returned as block elements cannot
+ * be nested into a list.
+ */
+@PublicEvolving
+public interface DescribedEnum {
+
+    /**
+     * Returns the description for the enum constant.
+     *
+     * <p>Typically, this will return a simple, single {@link TextElement}.
+     */
+    List<InlineElement> getDescription();
```

Comment:
Why do we use a `List` here? Is there a concrete use case?

Reply:
Yeah, if you want to add links or code, you'd need to use multiple inline elements.

Reply:
However, one could also wrap multiple inline elements into a single one, like I did in `ConfigOptionsDocGenerator#wrap`. Maybe it makes sense to simplify the API and just offer a utility method to do that instead.

Reply:
Yeah, I think that's better, since it's also the more common case. I've updated it.